### PR TITLE
feat: slack notifier on sign-ups [SUPA-14]

### PR DESCRIPTION
**Pull request by altaywtf**

### CODE REVIEW

**Comment by altaywtf** [comment]
Current dependencies on/for this PR:
* `main`
  * **PR #34** <a href="https://app.graphite.dev/github/pr/experiment-station/beecast/34?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #35** <a href="https://app.graphite.dev/github/pr/experiment-station/beecast/35?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #36** <a href="https://app.graphite.dev/github/pr/experiment-station/beecast/36?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/experiment-station/beecast/36?utm_source=stack-comment).

File: lib/services/spotify/save-user-info.ts
Comment (altaywtf) [inline lib/services/spotify/save-user-info.ts:1]: I don't want to check `isNewAccount` for everytime this runs but couldn't find if there's a better way to do this:
- upsert doesn't tell if it's `created` or `updated`
- we could check if `created_at` is less then current time minus some delta but it's ugly
- we could check if `created_at` is equal to `updated_at` but there's no `updated_at` in the account table

Comment (altaywtf) [reply lib/services/spotify/save-user-info.ts:1]: https://github.com/orgs/supabase/discussions/6741#discussioncomment-2713251 😭

Comment (altaywtf) [reply lib/services/spotify/save-user-info.ts:1]: ended up creating a naive check against `created_at`... would've added a db trigger / extension for the `updated_at` if it was bizniz critical but we just need this for notification so I guess it's ok 🤷‍♂️ 

https://github.com/experiment-station/beecast/pull/36/commits/0653032d7ed8a7a7130dd6834f494fc7374ce674

Comment (okanisildar) [reply lib/services/spotify/save-user-info.ts:1]: I think this looks nice. Another option is, changing upsert to find and insert or update. We can have more control for the flow, that way we can also get rid of `onConflic` part. BUT I am not sure if it is worth it 😝

File: lib/services/slack.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+import { ofetch } from 'ofetch';
+
+export const notify = async (message: string) => {
+  try {
+    await ofetch(env.SLACK_POSTMAN_WEBHOOK_URL, {
+      body: { text: message },
+      method: 'POST',
+    });
+  } catch (error) {
+    // Do nothing?
```
Comment (okanisildar) [inline lib/services/slack.ts:11]: yeah 😃

**Review by okanisildar — APPROVED** [review]

File: lib/services/slack.ts
Comment (okanisildar) [inline lib/services/slack.ts:1]: It's a nitpick but maybe we can move this to it's own folder. It'd be better to follow the convention. If it's a third party have the folder for files related to that third party, if it's from our db we can have it as a file. In addition, we can add more notifiers in the future (buy seats for instance 🤑). We might need more things to handle those (just a guess).

Comment (altaywtf) [reply lib/services/slack.ts:1]: deal 🤝

File: lib/services/slack.ts (hunk 1 above)
Comment (altaywtf) [reply lib/services/slack.ts:11]: guess we could log to `baselime` or `sentry` when we have em

File: lib/services/spotify/save-user-info.ts
Comment (altaywtf) [reply lib/services/spotify/save-user-info.ts:1]: 👍 let's leave it as it is then... 

we'll probably use `date-fns` in the UI as well 😬

File: lib/services/slack.ts
Comment (altaywtf) [reply lib/services/slack.ts:1]: [c07ce9b](https://github.com/experiment-station/beecast/pull/36/commits/c07ce9bcfeb12cf2d71df519f85b66432c16621b)

btw this PR depends on #39

**Comment by altaywtf** [comment]
gonna rebase

File: lib/services/notify/slack.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import { env } from '@/env.mjs';
+import { ofetch } from 'ofetch';
+
+export const notifyViaSlack = async (message: string) => {
```
Comment (altaywtf) [inline lib/services/notify/slack.ts:4]: not sure about the name

Comment (okanisildar) [reply lib/services/notify/slack.ts:4]: maybe `notifySlack`? In theory we're notifying slack :D

Comment (altaywtf) [reply lib/services/notify/slack.ts]: 👍

**Pull request merged**